src/main/java/com/learning/pages/PageBase.java

```diff
@@ -0,0 +1,12 @@
+package com.learning.pages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.PageFactory;
+
+public abstract class PageBase {
+  public PageBase(WebDriver driver) {
+    PageFactory.initElements(driver, this);
+  }
+
+  public abstract void waitForPageToLoad(WebDriver driver);
```

Comment:
This could be a method that does not take any parameter (webDriver field could be protected and reused).
This method could also return boolean to indicate if this page was actually loaded.